config/tsup.config.ts

```diff
@@ -0,0 +1,27 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { defineConfig } from 'tsup';
+
+const PACKAGE_ROOT_PATH = process.cwd();
+const INPUT_FILE_PATH = path.join(PACKAGE_ROOT_PATH, 'src/index.ts');
+const INPUT_FILE = fs.existsSync(INPUT_FILE_PATH)
+  ? INPUT_FILE_PATH
+  : path.join(PACKAGE_ROOT_PATH, 'src/index.tsx');
+
+export default defineConfig({
+  entry: [INPUT_FILE],
+  sourcemap: true,
+  clean: true,
+  format: ['cjs', 'esm'],
+  minify: false,
+  cjsInterop: true,
+  dts: true,
+  replaceNodeEnv: true,
+  skipNodeModulesBundle: true,
```

Comment:
Why is this required?

Reply:
this will skip bundling of anything imported that is not local to the package even if it is a dependency.
Ideally, for packages like these, I think it's best not to bundle node_module packages in the build files since it will be taken care of by the end user's bundler. 
Since we're not shipping browser bundles this provides some safety in my opinion.
If you think this is unnecessary I can remove it

Reply:
We will test it on next release